### PR TITLE
Branch 5.4 To support Report with Condition & Tranform

### PR DESCRIPTION
**Pull request by varun0002**
To support Report with Condition & Tranform.

Related to issue # 336

### CODE REVIEW

File: public/defaults/report_watcher.json [hunk 1]
```diff
@@ -6,6 +6,19 @@
     "schedule": {
       "later": "every 1 hour"
     }
+  },
```
Comment (sergibondarenko) [inline public/defaults/report_watcher.json]: Why do you add `input` and `condition` in the report action? The goal of the report is to take a screenshot of a web page. If you want to execute another action based on some condition in the same watcher, just add a new action under the same watcher.
screenshot from 2018-03-15 16-05-08
screenshot from 2018-03-15 16-05-25

**Comment by varun0002** [comment]
Kibana gets crash if default input & condition missing with watcher.js changes & that is due to following code line. This happens if someone directly creates new report which comes without any input & condition. 

     if (!request) {
        throw new Error(`Watcher search request is malformed: ${task._id}`);
      }
      if (!condition) {
        throw new Error(`Watcher condition is malformed: ${task._id}`);

Though, the goal is to take screenshot but it doesn't make sense to take screenshot of the dashboard which returns no results and send blank screenshot to user. For e.g. I have a dashboard to show hostname & count if my log message got "Error" in last 15 mins. User is only interested in this report if there' an error in given timeframe. However, someone can still use default input & condition even though they want blank screenshot.

**Comment by sergibondarenko** [comment]
@varun0002 I don't see Kibana v5.4.3 crashing when a report is executed, here is demo https://youtu.be/MYk9qMZ6tdI

The success console message:
```
  log   [10:13:20.446] [debug][status][Sentinl][email] Email sent. Watcher g0r931l5b3-9ta4mbucih8-4ui942mzd77: {"attachments":[{"path":"/tmp/report-d213312a-07a4-456b-8c3e-cea31b8409eb-16-03-2018-11-13-14.png","type":"image/png","name":"report-d213312a-07a4-456b-8c3e-cea31b8409eb-16-03-2018-11-13-14.png","headers":{"Content-ID":"<my-report>","Content-Type":"image/png; name=\"report-d213312a-07a4-456b-8c3e-cea31b8409eb-16-03-2018-11-13-14.png\"","content-disposition":"attachment; filename=cid:my-report"}}],"alternative":{"data":"<html><img src='cid:my-report' width='100%'></html>","alternative":true,"charset":"utf-8","type":"text/html","inline":true},"header":{"message-id":"<1521195200373.2.960@beast>","date":"Fri, 16 Mar 2018 11:13:20 +0100","from":"trex@beast.com","to":"sergibondarenko@email.com","subject":"=?UTF-8?Q?Sentinl_Report?="},"content":"text/plain; charset=utf-8","text":"Sample Sentinl Screenshot Report"}
```
But I understood what you want. You want to execute report only if certain condition met.

**Comment by varun0002** [comment]
@sergibondarenko: Kibana will crash with the recent changes in watcher.js (if there's no input & condition) which submitted in this pull request. Hence, modified report_watcher.json to create report alarm with default input & condition.

**Comment by sergibondarenko** [comment]
@varun0002 I created a new PR to fix several things. You can find your commits there https://github.com/sirensolutions/sentinl/pull/349

**Pull request closed without merging**